src/include/http_server.hpp

```diff
@@ -541,28 +543,32 @@ class HttpServer {
         server->_internal_event_handler(socket, events);
     }
 
-    std::unique_ptr< evhtp_ssl_cfg_t > get_ssl_opts_() {
+    static bool file_available(const char* filename){
         struct stat f_stat;
+        auto rc = ::stat(filename, &f_stat);
+        return !rc && f_stat.st_size;
+    }
+
+    std::unique_ptr< evhtp_ssl_cfg_t > get_ssl_opts_() {
         auto ssl_config{std::make_unique< evhtp_ssl_cfg_t >()};
 
         ssl_config->ssl_opts = 0; // SSL_OP_NO_SSLv2 | SSL_OP_NO_SSLv3 | SSL_OP_NO_TLSv1;
         ssl_config->pemfile = (char*)m_cfg.tls_cert_path.c_str();
         ssl_config->privfile = (char*)m_cfg.tls_key_path.c_str();
 
-        if (ssl_config->pemfile) {
-            if (::stat(ssl_config->pemfile, &f_stat) != 0) {
-                LOGERROR("Cannot load SSL cert: {}", ssl_config->pemfile);
-                return nullptr;
-            }
+        while (!file_available(ssl_config->pemfile)) {
+            LOGERROR("Cannot load SSL cert: {}! will try again in 10 seconds", ssl_config->pemfile);
```

Comment:
This should probably be an INFO or WARNING message since it's retried until it succeeds.